### PR TITLE
[21532] Apply new danger zone style

### PR DESCRIPTION
**Pull request by HDinger**
This applies the style for the new danger zone in:
- [x] Delete account/user (was done in the scope of https://github.com/opf/openproject/pull/3451)
- [x] Delete project
- [x] Delete work package
- [x] Delete Repository

WP: https://community.openproject.org/projects/openproject/work_packages/21532/activity

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hdinger-feature-21532-apply-danger-zone.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hdinger-feature-21532-apply-danger-zone.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/assets/stylesheets/content/_forms.sass [hunk 1]
```diff
@@ -67,36 +67,35 @@ $form--field-types: (text-field, text-area, select, check-box, radio-button, ran
     .form--section
       padding-top: 0px
 
+      & > *
+        padding-left: 1rem
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_forms.sass:72]: ~~Doesn't this have all kinds of non-danger-zony side effects?~~

edit: missed the `.danger-zone` selector above

**Comment by NobodysNightmare** [comment]
image

I think for work package deletion the button should also read "delete" and have the trash can icon. After all, this button will primarily delete things.

File: app/assets/stylesheets/content/_forms.sass [hunk 2]
```diff
@@ -67,36 +67,35 @@ $form--field-types: (text-field, text-area, select, check-box, radio-button, ran
     .form--section
       padding-top: 0px
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_forms.sass:68]: I just noticed that there is a marign at the bottom of the `form--section`. For the danger-zone you might want to fixate a `margin: 0`? IMO this looks more appealing...

File: config/locales/en.yml [hunk 3]
```diff
@@ -1322,6 +1322,14 @@ en:
   permission_view_wiki_edits: "View wiki history"
   permission_view_wiki_pages: "View wiki"
 
+  project:
+    destroy:
+      confirmation: "If you continue, the project %{identifier} and all related data will be permanently destroyed."
+      info: "Deleting the project is an irreversible action."
+      project_verification: "Enter the project's name %{name} to verify the deletion."
+      subprojects_confirmation: "Its subproject(s): %{value} will be also deleted."
```
Comment (NobodysNightmare) [inline config/locales/en.yml]: ... will also be deleted? (word swapping)

Not sure, though. I am not to confident about english grammar...

File: app/views/work_packages/bulk/destroy.html.erb [hunk 4]
```diff
@@ -45,29 +45,41 @@ See doc/COPYRIGHT.rdoc for more details.
...
         <%=styled_label_tag 'to_do_action_nullify', l(:text_assign_to_project) %>
         <div class="form--field-container">
           <%= styled_radio_button_tag 'to_do[action]' , 'nullify' %>
         </div>
       </div>
-      <div class="form--field">
-        <%= styled_label_tag 'to_do_action_reassign', l(:text_reassign) %>
-        <div class="form--field-container">
-          <%= styled_radio_button_tag 'to_do[action]', 'reassign', :onclick => 'if(jQuery("#to_do_action_reassign").prop("checked")) { jQuery("#to_do_reassign_to_id").focus(); }' %>
-          <%= styled_label_tag 'to_do_reassign_to_id', l(:text_reassign_to) %>
```
Comment (NobodysNightmare) [inline app/views/work_packages/bulk/destroy.html.erb:64]: You should not simply delete that label, since it carries contextual information for impaired users.

At the very least you should add an invisible label again (`class="hidden-for-sighted"`), that way a screen reader will read it out loud, when focussing that label (e.g. "ID; edit text" vs "work package with hint: ID; edit text" for ChromeVox).

It will also allow you to refer to that text field by a readable name in your cucumber tests.

edit: After consulting the [accessibility checklist](https://www.openproject.org/open-source/development/accessibility-checklist/), it might be sufficient to change the placeholder text to something like "work package ID".
Also thinking about it further the old navigation is not really great accessibility-wise, since it is not very clear which purpose the work package ID has in case you haven't selected the last option and jsut use your keyboard for navigation...

**tl; dr:** Some indication that the textfield expects a _work package_ ID would be nice, we won't need to make it perfectly accessible for now, because it wasn't before either.

File: spec/features/repositories/repository_settings_spec.rb [hunk 5]
```diff
@@ -62,9 +62,14 @@
...
-      expect(page).to have_selector(".notification-box.#{warning}")
-      find('a', text: I18n.t(:button_delete)).click
+      if type == 'managed'
+        expect(page).to have_selector('form.danger-zone')
+        find('.danger-zone .button').click
+      else
+        expect(page).to have_selector('.notification-box.-warning')
+        find('a', text: I18n.t(:button_delete)).click
+      end
+
```
Comment (NobodysNightmare) [inline spec/features/repositories/repository_settings_spec.rb:72]: Style: double empty line. Not sure why @houndci did not complain here...

File: spec/features/projects/delete_projects_spec.rb [hunk 6]
```diff
@@ -29,11 +29,11 @@
 require 'spec_helper'
 require 'features/projects/projects_page'
 
-describe 'Delete project', type: :feature do
+describe 'Delete project', type: :feature, js: true do
   let(:current_user) { FactoryGirl.create (:admin) }
   let(:project) { FactoryGirl.create(:project) }
   let(:projects_page) { ProjectsPage.new(project) }
-  let(:delete_button) { find('input[type="submit"]') }
+
```
Comment (NobodysNightmare) [inline spec/features/projects/delete_projects_spec.rb:36]: Style: double empty line. Not sure why @houndci did not complain here...
